dali/kernels/common/scatter_gather.cu

```diff
@@ -134,6 +143,20 @@ __global__ void BatchCopy(const ScatterGatherBase::CopyRange *ranges) {
   }
 }
 
+constexpr int kMaxRangesByVal = 2048 / sizeof(ScatterGatherBase::CopyRange);
+
+struct CopyRanges {
+  ScatterGatherBase::CopyRange ranges[kMaxRangesByVal];
+};
+
+__global__ void BatchCopy(CopyRanges ranges) {
+  auto range = ranges.ranges[blockIdx.x];
+
+  for (size_t i = threadIdx.x; i < range.size; i += blockDim.x) {
+    range.dst[i] = range.src[i];
```

Comment:
Side question. Does coalescing reads and writes here would help (into 32b values)?

Reply:
I've been thinking about just that - it'd certainly help. However, to make it viable, we'd need to have mutual alignment of the input and output - the kernel would grow in complexity significantly, it's certainly not something we can slip into this PR under the "optimize parameter passing" umbrella.